Peer/Program.cs

```diff
@@ -38,14 +39,33 @@ public static async Task<int> Main(string[] args)
             };
 
             Console.OutputEncoding = Encoding.UTF8;
-            
+
             var builder = new AppBuilder(new ServiceCollection())
                 .WithParseTimeServiceConfig(SetupParseTimeServices)
-                .WithSharedServiceConfig(SetupServices)
+                .WithSharedRuntimeConfig(SetupServices)
                 .WithConfiguration(configBuilder =>
                 {
-                    configBuilder.AddJsonFile(Constants.DefaultConfigPath, optional: true)
-                        .AddEnvironmentVariables();
+                    var configPath = Environment.GetEnvironmentVariable("PEER_CONFIGPATH")
+                                     ?? Constants.DefaultConfigPath;
+
+                    if (configPath.StartsWith("~/"))
+                    {
+
+                        configPath = Path.Join(
+                            Environment.GetFolderPath(Environment.SpecialFolder.UserProfile),
+                            configPath[2..]);
+                    }
+
+                    var editor = Environment.GetEnvironmentVariable("EDITOR");
+
+                    configBuilder.AddJsonFile(configPath, optional: true)
+                        .SetFileLoadExceptionHandler(x => x.Ignore = true)
+                        .AddEnvironmentVariables()
+                        .AddInMemoryCollection(new Dictionary<string, string?>
+                        {
+                            ["Peer:Environment:ConfigPath"] = configPath,
+                            ["Peer:Environment:Editor"] = editor
```

Comment:
Is there a risk here of overwriting existing config with empty values? Like if someone has editor in the config file or has config path in the peer__environment__ form then would setting null/empty from PEER_CONFIGPATH and/or EDITOR replace mask them with empty and cause us to revert to the default?

Reply:
Good point, should probably put this as the base layer instead

Reply:
f'd